base/distributed/macros.jl

```diff
@@ -48,6 +48,24 @@ macro fetchfrom(p, expr)
     :(remotecall_fetch($thunk, $(esc(p))))
 end
 
+# extract a list of modules to import from an expression
+extract_imports(x) = Symbol[]
+function extract_imports(ex::Expr)
```

Comment:
Not sure whether it's a big deal, but this might be more efficient as
```julia
function extract_imports!(imports, ex::Expr)
```
and having the `everywhere` caller execute it as `extract_imports!(Symbol[], ex)`. That way you don't have to allocate an array for every single token.